File: wiz-shop-java/src/main/java/com/example/wizshopjava/api/product/dto/ProductDetailResponse.java
package com.example.wizshopjava.api.product.dto;

import java.time.LocalDateTime;

import com.example.wizshopjava.domain.product.entity.Product;
import com.example.wizshopjava.domain.product.entity.ProductStatus;

public record ProductDetailResponse(
	String productName,
	Long price,
	String description,
	Long hit,
	ProductStatus status,
	LocalDateTime createdAt
) {

	public static ProductDetailResponse from(Product product) {
		return new ProductDetailResponse(
			product.getName(),
			product.getPrice(),
			product.getDescription(),
			product.getHit(),
			product.getStatus(),
			product.getCreatedAt()
		);
	}
}

File: wiz-shop/src/main/kotlin/com/example/wizshop/api/product/dto/ProductDetailResponse.kt
package com.example.wizshop.api.product.dto

import com.example.wizshop.domain.product.entity.Product
import com.example.wizshop.domain.product.entity.ProductStatus
import java.time.LocalDateTime

data class ProductDetailResponse(
    val productName: String,
    val price: Long,
    val description: String,
    val hit: Long,
    val status: ProductStatus,
    val sellerNickname: String,
    val createdAt: LocalDateTime
) {

    companion object {
        fun from(product: Product) = ProductDetailResponse(
            productName = product.name,
            price = product.price,
            description = product.description,
            hit = product.hit,
            status = product.status,
            sellerNickname = product.seller.nickname,
            createdAt = product.createdAt
        )
    }
}